lib/wanda/catalog.ex

```diff
@@ -138,49 +137,50 @@ defmodule Wanda.Catalog do
   defp find_custom_values(check_id, available_customizations) do
     Enum.find_value(available_customizations, [], fn
       %CheckCustomization{check_id: ^check_id, custom_values: custom_values} ->
-        custom_values
+        Enum.map(custom_values, &CustomizedValue.from_custom_value/1)
 
       _ ->
         nil
     end)
   end
 
-  defp map_selectable_check_values(custom_values, check_values, customization_globally_disabled?) do
-    Enum.map(check_values, fn %Value{
-                                name: value_name,
-                                default: default_value
-                              } = value ->
+  defp map_selectable_check_values(
+         custom_values,
+         check_values,
+         customization_globally_disabled?,
+         env
+       ) do
+    check_values
+    |> Evaluation.resolve_values(custom_values, env)
+    |> Enum.map(fn %ResolvedValue{spec: value_spec, name: value_name} = resolved_value ->
       %{
         name: value_name
       }
-      |> add_value_customizability(value, customization_globally_disabled?)
-      |> maybe_add_current_value(default_value)
-      |> maybe_add_customization(custom_values)
+      |> add_value_customizability(value_spec, customization_globally_disabled?)
+      |> maybe_add_original_value(resolved_value)
+      |> maybe_add_customization(resolved_value)
     end)
   end
 
-  defp maybe_add_current_value(%{customizable: false} = value, _), do: value
+  defp maybe_add_original_value(%{customizable: false} = value, _), do: value
 
-  defp maybe_add_current_value(%{customizable: true} = value, default_value),
-    do: Map.put(value, :current_value, default_value)
+  defp maybe_add_original_value(
+         %{customizable: true} = value,
+         %ResolvedValue{original_value: original_value}
+       ),
+       do: Map.put(value, :original_value, original_value)
 
-  defp maybe_add_customization(%{customizable: false} = value, _), do: value
+  defp maybe_add_customization(%{customizable: customizable} = value, %ResolvedValue{
```

Comment:
nitpick: is it not simpler to put this match after the next one without any `when`?
```
  defp maybe_add_customization(initial_value, _resolved_value), do: initial_value
```
I mean, if both customizable and customized are true we take the custom, otherwise we take the default value

Reply:
Well yes 😄 4a265c21d90a38ac543941d68ca9e92402684b52